sdk/nodejs/stackReference.ts

```diff
@@ -56,6 +57,25 @@ export class StackReference extends CustomResource {
     public getOutput(name: Input<string>): Output<any> {
         return all([output(name), this.outputs]).apply(([n, os]) => os[n]);
     }
+
+    /**
+     * Fetches the value promptly of the named stack output.  May return undefined if the value is
+     * not known for some reason.
+     *
+     * This operation is not supported (and will throw) if any exported values of the StackReference
+     * are secrets.
+     *
+     * @param name The name of the stack output to fetch.
+     */
+    public getOutputSync(name: string): any {
+        const out = this.getOutput(name);
+        const isSecret = promiseResult(out.isSecret);
+        if (isSecret) {
+            throw new Error("Cannot call [getOutputSync] on a StackReference containing secrets.  Only [getOutput] is supported.");
```

Comment:
Might be clearer to say "Cannot call [getOutputSync] if the referenced stack has secret outputs. Use [getOutput] instead."

Reply:
Sure!